runtime/src/accounts.rs

```diff
@@ -619,8 +619,13 @@ impl Accounts {
             .collect()
     }
 
-    pub fn calculate_capitalization(&self, ancestors: &Ancestors) -> u64 {
-        self.accounts_db.unchecked_scan_accounts(
+    pub fn calculate_capitalization(&self, ancestors: &Ancestors, slot: Slot) -> u64 {
+        let cap = self
+            .accounts_db
+            .update_accounts_hash_with_index_option(false, false, slot, ancestors, None, true)
+            .1;
+        // debug code for the moment
```

Comment:
this old code path remains for now to make sure we are getting the right results in all cases. I'll run this against mnb for a while. I am not yet familiar with all the ways calculate_capitalization is called. I'll look into it.

Reply:
so, to be clear I will not complete this pr with the old code path and the assert present.